src/rpc/claimtrie.cpp

```diff
@@ -22,143 +24,177 @@ uint160 ParseClaimtrieId(const UniValue& v, const std::string& strName)
     return result;
 }
 
+static CBlockIndex* BlockHashIndex(const uint256& blockHash)
+{
+    AssertLockHeld(cs_main);
+
+    if (mapBlockIndex.count(blockHash) == 0)
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found");
+
+    CBlockIndex* pblockIndex = mapBlockIndex[blockHash];
+    if (!chainActive.Contains(pblockIndex))
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Block not in main chain");
+
+    if (chainActive.Tip()->nHeight > (pblockIndex->nHeight + MAX_RPC_BLOCK_DECREMENTS))
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Block too deep to regenerate it");
+
+    return pblockIndex;
+}
 
 UniValue getclaimsintrie(const UniValue& params, bool fHelp)
 {
-    if (fHelp || params.size() > 0)
+    if (fHelp || params.size() > 1)
         throw std::runtime_error(
             "getclaimsintrie\n"
             "Return all claims in the name trie.\n"
             "Arguments:\n"
-            "None\n"
+            "1. \"blockhash\"     (string, optional) the hash of the block\n"
+            "                                           which is the active\n"
+            "                                           in the chain. If\n"
+            "                                           none is given, \n"
+            "                                           the latest block\n"
+            "                                           will be used.\n"
             "Result: \n"
```

Comment:
instead of "the hash of the block which is the active in the chain. If none is given, the latest block will be used", how about:

"get claims in the trie at the block specified by this block hash. If none is given, the latest active block will be used"

Reply:
getnameproof says "Block too deep to generate proof"